lxd/db/cluster/secrets.go

```diff
@@ -0,0 +1,270 @@
+package cluster
+
+import (
+	"context"
+	"crypto/rand"
+	"database/sql"
+	"database/sql/driver"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/canonical/lxd/lxd/db/query"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/entity"
+)
+
+// authSecretLength defines the length of an AuthSecretValue.
+const authSecretLength = 64
+
+// AuthSecretValue is used to store byte slice of length authSecretLength in a text column as a base64 encoded string.
+type AuthSecretValue []byte
+
+// newAuthSecretValue returns a AuthSecretValue, read from [rand.Reader].
+func newAuthSecretValue() AuthSecretValue {
+	secretValue := make([]byte, authSecretLength)
+	_, _ = rand.Read(secretValue)
+	return secretValue
+}
+
+// Validate checks that the AuthSecretValue has length authSecretLength.
+func (s AuthSecretValue) Validate() error {
+	if len(s) != authSecretLength {
+		return fmt.Errorf("Secret must have length %d", authSecretLength)
+	}
+
+	return nil
+}
+
+// Value implements [driver.Valuer] for AuthSecretValue.
+func (s AuthSecretValue) Value() (driver.Value, error) {
+	err := s.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	return base64.StdEncoding.EncodeToString(s), nil
+}
+
+// ScanText implements [query.TextScanner] for AuthSecretValue to simplify the [sql.Scanner] implementation.
+func (s *AuthSecretValue) ScanText(str string) error {
+	out, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		return err
+	}
+
+	value := AuthSecretValue(out)
+	err = value.Validate()
+	if err != nil {
+		return err
+	}
+
+	*s = value
+	return nil
+}
+
+// Scan implements [sql.Scanner] for AuthSecretValue.
+func (s *AuthSecretValue) Scan(value any) error {
```

Comment:
Should this type have a `String()` function to print its human readable form?

Reply:
Yep sounds good.